test/test_fsdp.py

```diff
@@ -97,12 +100,21 @@ def fsdp_main(rank, world_size, args):
     bsz_local_end = int((rank + 1) / world_size * B)
     ref_input_local = ref_input_global[bsz_local_start:bsz_local_end].to(rank)
 
-    for _ in range(N_ITER):
+    def forward_backward():
         optimizer.zero_grad()
         y_local = model(ref_input_local)
         y_local.sum().backward()
         sync_float8_amax_and_scale_history(model)
         optimizer.step()
+        return y_local
+
+    # TODO: Fix here. Running for one iteration and then compile is a workaround for compiling fsdp float8.
+    # A better way for compile is to FSDP(torch.compile(model)).
+    forward_backward()
+    if COMPILE:
+        forward_backward = torch.compile(forward_backward)
```

Comment:
are we sure this is supported?

how about something simpler such as

```
model = torch.compile(model)
sync_func = torch.compile(sync_float8_amax_and_scale_history)
for _ in range(N_ITER):
   ... loop as before ...
```

Reply:
This did work in fsdp.py test. But let me change it to the officially supported `FSDP(torch.compile(model))`.